server/pulp/server/controllers/importer.py

```diff
@@ -0,0 +1,281 @@
+import logging
+import sys
+
+import celery
+from mongoengine import ValidationError
+
+from pulp.common import error_codes, tags
+from pulp.plugins.config import PluginCallConfiguration
+from pulp.plugins.loader import api as plugin_api
+from pulp.server import exceptions
+from pulp.server.async.tasks import Task
+from pulp.server.db import model
+from pulp.server.managers import factory as manager_factory
+
+
+_logger = logging.getLogger(__name__)
+
+
+def build_resource_tag(repo_id, importer_type_id):
+    """
+    Returns an identfifier for the repo and importer.
+
+    :param repo_id: unique ID for a repository
+    :type  repo_id: basestring
+    :param importer_type_id: unique ID for the importer
+    :type  importer_type_id: basestring
+
+    :return: a globally unique identifier for the repo and importer that
+             can be used in cross-type comparisons.
+    :rtype:  basestring
+    """
+    RESOURCE_TEMPLATE = 'pulp:importer:%s:%s'
+    return RESOURCE_TEMPLATE % (repo_id, importer_type_id)
+
+
+def clean_config_dict(config):
+    """
+    Remove keys from a dict that have a value None.
+
+    :param config: configuration for plugin
+    :type  config: dict
+
+    :return: config without the keys whose values were None
+    :rtype:  dict:
+    """
+
+    if config is not None:
+        return dict([(k, v) for k, v in config.items() if v is not None])
+    else:
+        return None
+
+
+@celery.task(base=Task, name='pulp.server.managers.repo.importer.set_importer')
+def set_importer(repo_id, importer_type_id, repo_plugin_config):
```

Comment:
It looks like this function only gets called from one place, and that place has the repo object. It could pass in that object, instead of making this function look it up below.